src/main/scala/io/viash/helpers/IO.scala

```diff
@@ -126,7 +128,18 @@ object IO {
       }
     } else if (uri.getScheme == "http" || uri.getScheme == "https") {
       val url = new URL(uri.toString)
-      (url #> path.toFile).!!
+
+      // Prevent hanging when resource is not available
+      val connection = url.openConnection().asInstanceOf[HttpURLConnection]
+      connection.setConnectTimeout(5000)
+      connection.setReadTimeout(5000)
+      connection.connect()
+
+      if (connection.getResponseCode >= 400)
+        throw new RuntimeException(s"Could not download file: ${uri.toString} , Response code: ${connection.getResponseCode}")
+      else
+        (url #> path.toFile).!!
```

Comment:
Can we use the connection to download the file? Or is this happening implicitly already? (Seems unlikely to me)

Reply:
new version: